### ISSUE TITLE
RandomSurvivalForest integration with sklearn.GridSearchCV

### ISSUE THREAD

**Issue by EWolt** [sebp/scikit-survival#150]
I've really liked using this package for Survival Analysis, implementation is as straightforward as described and results are easily obtained. I'm however struggling with combining it with a grid search from scikit-learn. Especially on the following two parts:
- the target features (event and time_to_event) from the structured array format that is required, are conceptually tricky (at least for me) to map to a scorer metric in the scikit-learn package; and
- the Random Survival Forest, that I'm using currently, doesn't create an output that is useable by the GridSearchCV object's classifier metrics.

Would you consider including a version of GridSearchCV in the scikit-survival package as to more easily apply it to to the models in the package? 

I've been looking into writing a custom scorer function for the concordance index (as described in https://codereview.stackexchange.com/questions/202140/concordance-index-calculation) but as my skill level in programming in Python is pretty basic this will probably be a very long process.  Additionally, I can imagine that a lot of users would find this feature very helpful!

**Comment by sebp** [comment]
There's nothing special you need to consider. sklearn can deal with the 
structured array for y, and each survival model's score method, which is 
used by GridSearchCV by default, returns the c-index.

Have a look at the example combining GridSearchCV with 
CoxnetSurvivalAnalysis, which you can adapt for your needs:
https://gist.github.com/sebp/d580d44c4beab3379c6dfda6810b33b8

**Comment by EWolt** [comment]
Thanks! This worked perfectly indeed, how would you suggest approaching the use of other scoring metrics that you've included in sksurv (such as the cumulative_dynamic_auc) in a gridsearch?

**Comment by sebp** [comment]
I see two ways.

1. Write a wrapper of `cumulative_dynamic_auc` that always passes the full data, before splitting into train/test, as the `survival_train` and forward whatever is `GridSearchCV` passes you as `survival_test` and `estimate`.
2. You can write a wrapper class that looks like a regular sklearn estimator (with fit, predict, score). In the fit method, you can create a copy of training data and use it in the score method when calling `cumulative_dynamic_auc`.

I guess it would be nice if some of these ideas could be included into sksurv for ease of use.
